diff --git a/ml/backend/ggml/ggml/include/ggml.h b/ml/backend/ggml/ggml/include/ggml.h
index 20c912d..f42437b 100644
--- a/ml/backend/ggml/ggml/include/ggml.h
+++ b/ml/backend/ggml/ggml/include/ggml.h
@@ -422,7 +422,30 @@ extern "C" {
         // GGML_TYPE_IQ4_NL_4_8 = 37,
         // GGML_TYPE_IQ4_NL_8_8 = 38,
         GGML_TYPE_MXFP4   = 39, // MXFP4 (1 block)
-        GGML_TYPE_COUNT   = 40,
+
+        GGML_TYPE_TBQ3_0  = 42,
+        GGML_TYPE_TBQ4_0  = 43,
+        GGML_TYPE_TBQP3_0 = 44,
+        GGML_TYPE_TBQP4_0 = 45,
+        GGML_TYPE_TBQ3_1  = 46,
+        GGML_TYPE_TBQ4_1  = 47,
+        GGML_TYPE_TBQP3_1 = 48,
+        GGML_TYPE_TBQP4_1 = 49,
+        GGML_TYPE_TBQ3_2  = 50,
+        GGML_TYPE_TBQ4_2  = 51,
+        GGML_TYPE_TBQP3_2 = 52,
+        GGML_TYPE_TBQP4_2 = 53,
+        GGML_TYPE_TBQ3_3  = 54,
+        GGML_TYPE_TBQ4_3  = 55,
+        GGML_TYPE_TBQP3_3 = 56,
+        GGML_TYPE_TBQP4_3 = 57,
+        GGML_TYPE_TBQ3_4  = 58,
+        GGML_TYPE_TBQ4_4  = 59,
+        GGML_TYPE_TBQP3_4 = 60,
+        GGML_TYPE_TBQP4_4 = 61,
+        GGML_TYPE_AMX3_1  = 62,
+        GGML_TYPE_AMXV3_1 = 63,
+        GGML_TYPE_COUNT   = 64,
     };
 
     // precision
